src/background/browser-message-broadcaster-factory.ts

```diff
@@ -14,9 +14,7 @@ export class BrowserMessageBroadcasterFactory {
         // condition described by #1816
         await this.sendMessageToFrames(message);
 
-        const allTabs: chrome.tabs.Tab[] = await new Promise(resolve => {
-            this.browserAdapter.tabsQuery({}, resolve);
-        });
+        const allTabs = await this.browserAdapter.tabsQuery({});
```

Comment:
Should we have a type for `allTabs`?

Reply:
That was necessary before because by using the promise to wrap a call to the callback-based `tabsQuery` we get a result type of `unknown` (even though the callback of that function does have proper typing), hence, we need to be explicit with the types. If we don't, we get type errors like it's shown in this screenshot
![image](https://user-images.githubusercontent.com/2837582/71425386-fc476d00-2650-11ea-8e7b-0c5d798c3cf3.png).
By changing the signature of `tabsQuery` to explicitly return a `Promise<Tabs.Tab[]>` the need to be explicit goes away, as the function is already being explicit with the return type and typescript is able to assign the proper type to `allTabs`.